clang/lib/AST/Decl.cpp

```diff
@@ -4170,7 +4170,7 @@ FunctionTemplateDecl *FunctionDecl::getPrimaryTemplate() const {
   if (FunctionTemplateSpecializationInfo *Info
         = TemplateOrSpecialization
             .dyn_cast<FunctionTemplateSpecializationInfo*>()) {
-    return Info->getTemplate();
+    return Info->getTemplate()->getMostRecentDecl();
```

Comment:
why `mostRecent` vs the definition?  Should we prefer the definition?

Reply:
We treat member specializations as redeclarations of the instantiated declaration. Using the definition would cause problems if the member specialization isn't a definition but the specialized declaration is, e.g.

```cpp
template<typename T>
struct A
{
    template<bool U>
    static constexpr bool f() requires U
    {
        return true;
    }
};

template<>
template<bool U>
constexpr bool A<short>::f() requires U;
```

Reply:
Ah!  Got it, thanks.